backend/clubs/views.py

```diff
@@ -506,6 +517,37 @@ def get_serializer_class(self):
             return WritableClubFairSerializer
         return ClubFairSerializer
 
+    @action(detail=True, methods=["get"])
+    def get_events_user_info(self):
+        """
+        Returns all events, grouped by id, with the number of participants currently in the meeting,
+        the officers or owners in the meeting, and the number of participants
+        who have already attended the meeting
+        """
+        fair = self.get_object()
+        clubs = fair.participating_clubs.all()
+        events = Event.objects.filter(
+            club__in=clubs,
+            type=Event.FAIR,
+            start_time_gte=fair.start_time,
+            end_time_lte=fair.end_time,
+        )
+
+        response = [
+            {
+                event.id: {
+                    "num_participants": len(
+                        ZoomMeetingVisit.objects.filter(event=event, leave_time=None)
+                    ),
+                    "officers": get_officers(event),
+                    "already_attended": len(ZoomMeetingVisit.objects.filter(event=event)),
+                }
+            }
+            for event in events
+        ]
```

Comment:
You want to return a dictionary, not a list of dictionaries. Your current output generates something like this:

```
[{3: ...}, {4: ...}, {5: ...}]
```

The desired result is this:

```
{3: ..., 4: ..., 5: ...}
```

You can do list comprehensions with dictionaries, for example you can try running `{a: a for a in range(3)}` which will produce `{0: 0, 1: 1, 2: 2}`.

Reply:
woops sorry, I had it as something like "event" : event.id  : {num_part, officers, already..} before but I removed the "event" key because it was redundant